maven-core/src/main/java/org/apache/maven/graph/DefaultProjectDependencyGraph.java

```diff
@@ -59,6 +66,16 @@ public DefaultProjectDependencyGraph( Collection<MavenProject> projects )
         super();
         this.allProjects = Collections.unmodifiableList( new ArrayList<>( projects ) );
         this.sorter = new ProjectSorter( projects );
+        this.order = new HashMap<>();
+        this.projects = new HashMap<>();
+        List<MavenProject> sorted = this.sorter.getSortedProjects();
+        for ( int index = 0; index < sorted.size(); index++ )
+        {
+            MavenProject project = sorted.get( index );
```

Comment:
This should rather use an iterator style and a separately incremended index. Generally, this will incur `O(n)` and is antipattern.

Reply:
Yeah, well, the list is backed by an `ArrayList` in this case and not a `LinkedList`, but the change is ok.

Reply:
That is likely true, but the `ArrayList` is an implementation detail and shouldn't really matter. So don't make this assumption.